### PR TITLE
Make it possible to set the LDAP_BASE_DN via env

### PR DESCRIPTION
**Pull request by EugenMayer**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you updated the readme?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss open a ticket first**

This is backwards compatible, not setting the value means use the LDAP_DOMAIN - as it was before

### CODE REVIEW

**Comment by EugenMayer** [comment]
I understand that there is this

```
LDAP_BASE_DN: Ldap base DN. If empty automatically set from LDAP_DOMAIN value. Defaults to (empty)
```

But i suppose having those differen (like we have) is not very unusual. Being able to setup them seperatly

**Comment by EugenMayer** [comment]
After testing it a bit i came to the conlusion

- this change is backwards compatible, means, that not setting the value, which is/was the default, will fallback to `LDAP_DOMAIN` which was the default, and is the default now
- while setting the LDAP_BASE_DN is possible, there is a known glitch. Assuming one using `foo.com` as the LDAP_DOMAIN but `dc=sub,dc=foo,dc=com` as the LDAP_BASE_DN, the actual admin user is created under `admin.foo.com`, not admin.sub.foo.com`

After understanding my old setup i seem to have confused the LDAP_DOMAIN with the "expected domain the user connects to the ldap server", so basically i understood is as the exposed hostname when clients connect to the ldap server - but this is not the case, the server does not have such a field/header, the client has.

So in the end, i'am just able to switch the actual LDAP_DOMAIN to the LDAP_BASE_DN and thus ensure there is no glitching with the admin user.

Bottom line, offering this setting invites to probably a "flexible" setup, which i'am currently not aware why it would be needed, but it introduces the opportunity for suprising issues.

That said, form my POV, that PR should probably not get merged
